### ISSUE TITLE
Excess strike-through on conversion

### ISSUE THREAD

**Issue by mikemccand** [apache/lucene-jira-archive#89]
GitHub issue: https://github.com/mocobeta/forks-migration-test/issues/7230
Jira issue: https://issues.apache.org/jira/browse/LUCENE-7260

Towards the end of the opening description in GitHub there is this text:

```
For Lucene 3.6.2, the timings settle down to 200300 with the fastest being 207.
For Lucene 5.4.1, the timings settle down to 2000030000 with the fastest being 22444.
```

But it has large strike-through section that the original Jira does not have.

I think it is because `~` is being interpreted as MarkDown and is not escaped on migration?

**Comment by mikemccand** [comment]
OK I confirmed that we can just escape the `~` with a backslash:

Broken (copied from the Jira issue):

For Lucene 3.6.2, the timings settle down to 200~300 with the fastest being 207.
For Lucene 5.4.1, the timings settle down to 20000~30000 with the fastest being 22444.

Fixed:

For Lucene 3.6.2, the timings settle down to 200\~300 with the fastest being 207.
For Lucene 5.4.1, the timings settle down to 20000\~30000 with the fastest being 22444.

But we must do this after remapping accounts!

**Comment by mocobeta** [comment]
I know which code we should change - will take a look.

**Comment by mocobeta** [comment]
I found this is trickier than it looks...

**Comment by mocobeta** [comment]
The difficulty here is that `~` actually has special meanings in both Jira syntax and Markdown.

The conversion process is done as follows and the order is important.

1. Convert Jira syntax to Markdown
2. Convert Jira mentions to GitHub mentions

Escaping `~` should be done _before_ 1) syntax conversion; otherwise valid strike-throughs in Markdown are affected. However, it also should be done _after_ 2) mention conversion; otherwise `[~username]` mentions are affected. 

We could find a tricky workaround, but I'm afraid any workaround has unpredictable side effects - I would leave this issue.

**Comment by mocobeta** [comment]
No, in fact, it should be safely done - #108.

**Comment by mocobeta** [comment]
Fixed in #108.
